### DISCUSSION TITLE
oo/extends-bigsai

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#103] [Announcements]
# oo/extends-bigsai

Java程序员进阶之路，小白的零基础Java教程，认真聊聊 Java的三大特征：继承

https://tobebetterjavaer.com/oo/extends-bigsai.html

**Comment by wylgithub** [comment]
感谢分享

**Comment by 2258324319** [comment]
父类的静态方法不能被子类重写为非静态方法，这一条对接口应当不适用吧，毕竟接口都是public static

**Comment by 2258324319** [comment]
很全面，对着这个自己再深度遍历一下知识点就7788了

**Reply by itwanger** [reply]
👍🏻

**Comment by dark-eternity** [comment]
小黑子漏出鸡脚了

**Comment by dikk123** [comment]
而在创建子类对象的时候需要先创建父类对象，所以父类优先于子类。====创建子类对象的时候，只是调用了父类的构造防范，并没有创建父类对象。
